### ISSUE TITLE
Keycloak-connect deprecated, use another middleware ?

### ISSUE THREAD

**Issue by Zied-ncq** [ferrerojosh/nest-keycloak-connect#152]
Since the release of Keycloak 19, there were a plan to deprecate all keycloak connect adapters including keycloak-connect.

- https://www.npmjs.com/package/keycloak-connect
- https://www.keycloak.org/2023/03/adapter-deprecation-update.html#:~:text=In%202022%20we%20announced%20the,most%20adapters%20in%20Keycloak%2019.

Is there a way to include another Oauth2 middleware like passport or any other provider.

**Comment by kasir-barati** [comment]
Any update on this?

**Comment by ferrerojosh** [comment]
Yeah I think we could use another middleware, but I'll need to find an alternative for keycloak's authz feature. I think there should be an existing api for that, was it called keycloak-admin or something.

**Comment by kasir-barati** [comment]
> Yeah I think we could use another middleware, but I'll need to find an alternative for keycloak's authz feature. I think there should be an existing api for that, was it called keycloak-admin or something.

I guess you're talking about https://github.com/keycloak/keycloak/tree/main/js/libs/keycloak-admin-client

**Comment by valerii15298** [comment]
Also there is another interesting alternative which I proposed:

https://github.com/keycloak/keycloak-nodejs-connect/issues/492

**Comment by devloe** [comment]
Any updates?
